CMakeLists.txt

```diff
@@ -1,10 +1,149 @@
-cmake_minimum_required(VERSION 3.10)
+# ------------------------------------------------------------------------- #
+# Application for all the components needed for the GDAS system             #
+# ------------------------------------------------------------------------- #
 
-# set the project name
-project(GDASApp)
+# Check for minimim cmake requirement
+cmake_minimum_required( VERSION 3.12 FATAL_ERROR )
 
-find_package(Python3 REQUIRED COMPONENTS Interpreter)
+find_package(ecbuild 3.5 REQUIRED HINTS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild)
+
+project(GDASApp VERSION 1.0.0 LANGUAGES C CXX Fortran )
 
+include(GNUInstallDirs)
 enable_testing()
 
+# Build type.
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# Find dependencies.
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+# Include ecbuild_bundle macro
+include( ecbuild_bundle )
+
+# Enable MPI
+set( ENABLE_MPI ON CACHE BOOL "Compile with MPI" )
+
+# Handle user options.
+option(BUILD_GDASBUNDLE "Build GDAS Bundle" ON)
+
+# Initialize bundle
+# -----------------
+ecbuild_bundle_initialize()
+
+# Build bundle source code.
+if(BUILD_GDASBUNDLE)
+
+# jedi-cmake
+  ecbuild_bundle( PROJECT jedicmake GIT "https://github.com/jcsda/jedi-cmake.git" BRANCH develop UPDATE )
```

Comment:
I tried `add_subdirectory(src)` and then retain all this in `src/CMakeLists.txt`.  See https://github.com/NOAA-EMC/GDASApp/commit/0ec3fb472a7f77abd1a376a40ee6da1dbd1349ce .  But, I couldn't get it to work. 
This will need a resolution eventually, since NCO will not want to see all the bundle specific repos at the top-level.